helm/charts/determined/templates/master-config.yaml

```diff
@@ -246,28 +246,32 @@ stringData:
       log_policies:
         {{- toYaml .Values.taskContainerDefaults.logPolicies | nindent 8}}
       {{- end }}
-      {{- if and .Values.taskContainerDefaults.cpuImage .Values.taskContainerDefaults.gpuImage }}
       image:
+        {{- if .Values.taskContainerDefaults.cpuImage }}
          cpu: {{ .Values.taskContainerDefaults.cpuImage | quote }}
-         gpu: {{ .Values.taskContainerDefaults.gpuImage | quote }}
-      {{- else }}
-      image:
+        {{- else }}
          cpu: {{ .Values.defaultImages.cpuImage | quote }}
+        {{- end }}
+        {{- if .Values.taskContainerDefaults.gpuImage }}
+         gpu: {{ .Values.taskContainerDefaults.gpuImage | quote }}
+        {{- else }}
          gpu: {{ .Values.defaultImages.gpuImage | quote }}
-      {{- if or .Values.taskContainerDefaults.cpuImage .Values.taskContainerDefaults.gpuImage }}
-        {{ required "A valid .Values.taskContainerDefaults.cpuImage entry is required if setting .Values.taskContainerDefaults.gpuImage!" .Values.taskContainerDefaults.cpuImage }}
```

Comment:
we are removing this required check meaning you no longer need to set both a gpu / cpu image right?

I think someones cluster will likely not work if they don't have both a cpu and gpu image specified

Reply:
Yeah. In this rewrite, there is always a `cpu` and `gpu` image. Whichever one isn't set will be inferred from `.Values.defaultImages`. (same goes for `rocm`).